python_modules/dagster/dagster/core/host_representation/external_data.py

```diff
@@ -510,9 +551,14 @@ def external_partition_set_data_from_def(partition_set_def):
 def external_sensor_data_from_def(sensor_def):
     return ExternalSensorData(
         name=sensor_def.name,
-        pipeline_name=sensor_def.pipeline_name,
-        solid_selection=sensor_def.solid_selection,
-        mode=sensor_def.mode,
```

Comment:
when exactly does `keep legacy field populated` flow kick in, just when reading? Should we still be writing these out in the single target case?